util/dvsim/dvsim.py

```diff
@@ -75,14 +73,12 @@ def resolve_scratch_root(arg_scratch_root):
     else:
         arg_scratch_root = os.path.realpath(arg_scratch_root)
 
-    try:
-        os.system("mkdir -p " + arg_scratch_root)
-    except OSError:
-        log.fatal(
-            "Invalid --scratch-root=\"%s\" switch - failed to create directory!",
-            arg_scratch_root)
+    if arg_scratch_root in ["/", "/root"]:
+        log.fatal("Scratch cannot be set to system root directory.")
```

Comment:
This doesn't look right to me! I guess this is in response to [a message I left on the previous version of this patch set](https://github.com/lowRISC/opentitan/pull/5203#discussion_r579179927) where I pointed out that it would be good to catch errors in case someone set the scratch root to `/root`. I wasn't clear. I mean that it would be good to catch errors if someone has set the scratch root to something that isn't writable: there's nothing particularly special about the `/root` directory.

Reply:
Aah should have paid a bit more attention to detail here - my mistake. Fixed.